ext/standard/http_fopen_wrapper.c

```diff
@@ -726,6 +726,11 @@ static php_stream *php_stream_url_wrap_http_ex(php_stream_wrapper *wrapper,
 			}
 			ZVAL_STRINGL(&http_response, tmp_line, tmp_line_len);
 			zend_hash_next_index_insert(Z_ARRVAL_P(response_header), &http_response);
+		} else {
+			php_stream_close(stream);
+			stream = NULL;
```

Comment:
While merging this up into 7.4, I found that we have https://github.com/php/php-src/commit/41c4816157bc0201b30877591fb595686d665748 there, which should have much the same effect -- the main difference I see is that it does not close and null the stream. The EOF branch below also doesn't do this. I'm wondering if it's necessary?

Reply:
I did this to be coherent with lines 851 & 852 (https://github.com/php/php-src/blob/86e6cffc358a3982bfa99ce65e63ad76a3299b30/ext/standard/http_fopen_wrapper.c#L851)

Removing both lines doesn't work because the stream is not technically EOF, so it still go through the second call, thus applying 2 times the timeout.

Only `php_stream_close(stream)` -> seg fault
Only `stream = null` -> seems to work